src/heuristic/HeuristicMapper.cpp

```diff
@@ -301,21 +317,15 @@ void HeuristicMapper::createInitialMapping() {
 }
 
 void HeuristicMapper::mapUnmappedGates(
-    std::size_t layer, std::vector<std::uint16_t>& consideredQubits) {
-  for (const auto& gate : layers.at(layer)) {
-    if (gate.singleQubit()) {
-      continue;
-    }
+    const TwoQubitMultiplicity& twoQubitGateMultiplicity) {
+  for (const auto& edgeMultiplicity : twoQubitGateMultiplicity) {
+    auto q1 = edgeMultiplicity.first.first;
+    auto q2 = edgeMultiplicity.first.second;
```

Comment:
You could benefit from using structured bindings here. Something like
```suggestion
  for (const auto& [edge, multiplicity] : twoQubitGateMultiplicity) {
    const auto& [q1, q2] = edge;
```
You might require a `[[maybe_unused]]` for the `multiplicity` binding, or try to get around the unused warning by using `_` as a name instead, which should also work, i.e., 
```c++
  for (const auto& [edge, _] : twoQubitGateMultiplicity) {
    const auto& [q1, q2] = edge;
```

Reply:
Great suggestion. Makes the code much cleaner. Thank you!